traitsui/editors/video_editor.py

```diff
@@ -39,50 +40,59 @@ class VideoEditor(BasicEditorFactory):
+    #: reference to a trait that will synchronize to the volume.
     #: Values are on a logarithmic scale.
-    volume = Str(sync_value='from', sync_name='volume')
-
-    #: The name of a trait to synchronise with the playback rate of
-    #: the video.  The referenced trait should be a Float.
-    playback_rate = Str(sync_value='from', sync_name='playback_rate')
-
-    #: The name of a trait to synchronise with the player's state.
-    #: The referenced trait should be a PlayerState.
-    state = Str(sync_value='both', sync_name='state')
-
-    #: The name of a trait to synchronise with the player's position.
-    #: The referenced trait should be a Float representing time in seconds.
-    position = Str(sync_value='both', sync_name='position')
-
-    #: The name of a trait to synchronise with the player's duration.
-    #: The referenced trait should be a Float representing time in seconds.
-    duration = Str(sync_value='to', sync_name='duration')
-
-    #: The name of a trait to synchronise with the player's media status.
-    #: The referenced trait should be a MediaStatus.
-    media_status = Str(sync_value='to', sync_name='media_status')
-
-    #: The name of a trait to synchronise with the player's buffer status.
-    #: The referenced trait should be an Range(0, 100).
-    buffer = Str(sync_value='to', sync_name='buffer')
-
-    #: The name of a trait to synchronise with the player's error state.
-    #: The referenced trait should be a Str.
-    video_error = Str(sync_value='to', sync_name='video_error')
-
-    #: The name of a trait to synchronise with the player's image function.
-    #: The referenced trait should be a Str.
-    image_func = Str(sync_value='both', sync_name='image_func')
-
-    #: The name of a trait to synchronise with the player's notify interval.
-    #: The referenced trait should be a Float representing time in seconds.
-    notify_interval = Str(sync_value='from', sync_name='notify_interval')
+    volume = CVType(Range(0.0, 100.0), sync_value='from')
+
+    #: The default playback rate of the video player or a ContextValue
+    #: reference to a trait that will synchronize to the playback rate.
+    playback_rate = CVType(Float, sync_value='from')
+
+    #: The initial player state of the video player or a ContextValue
+    #: reference to a trait that will synchronize to the player state.
+    state = CVType(PlayerState, sync_value='both')
+
+    #: The initial playback position of the video player or a ContextValue
+    #: reference to a trait that will synchronize to the playback position.
+    #: The value is a floating point time in seconds.
+    position = CVType(Float, sync_value='both')
+
+    #: A ContextValue reference to a trait that will synchronize to the
+    #: duration of the video.  The value is a floating point time in
+    #: seconds.
+    duration = Instance(ContextValue, args=(), allow_none=False,
+                        sync_value='to')
```

Comment:
`args` can't be empty, otherwise you get an exception when nothing is passed to the editor factory.

Reply:
Arg.  Should probably be `('',)` there and elsewhere.  Or maybe want to fix the definition of `ContextValue`.